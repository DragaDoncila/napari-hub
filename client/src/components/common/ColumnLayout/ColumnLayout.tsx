import clsx from 'clsx';
import { HTMLAttributes, ReactNode } from 'react';

interface Props extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
}

export function ColumnLayout({ children, ...props }: Props) {
  return (
    <div
      className={clsx(
        // Layout
        'flex 2xl:grid 2xl:justify-center',

        // Padding
        'p-6 md:p-14 2xl:px-0',

        // Grid gap
        '2xl:gap-14',

        // Grid columns
        '2xl:grid-cols-napari-2-col 3xl:grid-cols-napari-3-col',
      )}
      {...props}
    >
      {children}
    </div>
  );
}
