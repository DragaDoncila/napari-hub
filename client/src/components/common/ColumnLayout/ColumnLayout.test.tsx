import { render } from '@testing-library/react';

import { ColumnLayout } from './ColumnLayout';

function renderLayout() {
  const component = render(
    <ColumnLayout>
      <div />
      <div />
      <div />
    </ColumnLayout>,
  );

  return { component };
}

describe('<ColumnLayout />', () => {
  it('should match snapshot', () => {
    const { component } = renderLayout();
    expect(component.asFragment()).toMatchSnapshot();
  });
});
