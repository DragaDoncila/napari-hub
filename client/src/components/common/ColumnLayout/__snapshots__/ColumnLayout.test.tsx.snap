// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColumnLayout /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex 2xl:grid 2xl:justify-center p-6 md:p-14 2xl:px-0 2xl:gap-14 2xl:grid-cols-napari-2-col 3xl:grid-cols-napari-3-col"
  >
    <div />
    <div />
    <div />
  </div>
</DocumentFragment>
`;
